text/0000-additional-app-metadata.md

```diff
@@ -0,0 +1,130 @@
+# Meta
+[meta]: #meta
+- Name: Additional Application Image Metadata
+- Start Date: 2024-03-05
+- Author(s): dmikusa
+- Status: Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+When using a `Dockerfile`, it is possible to set additional metadata items like exposed ports and health check information. This metadata can be used by external systems, like container orchestrators, to automatically configure and deploy application images. It is not presently possible to set this metadata using buildpacks.
+
+# Definitions
+[definitions]: #definitions
+
+Dockerfiles settings for the proposed metadata:
+
+- Ports can be set with [`EXPOSE`](https://docs.docker.com/reference/dockerfile/#expose)
```

Comment:
You motivate supporting `EXPOSE` by talking about orchestration systems that will use the information in  `EXPOSE`.  Could you provide a concrete example?  I'm wondering if `HEALTHCHECK` can be supported without needing to also support `EXPOSE`.

Reply:
I don't use it myself, but someone I was talking (the spark for this RFC), made it seems like Google Cloud Run does things that way. Docker/Docker Compose also do, to a lesser extent. If you export that metadata, I believe you can `docker run -P` and it just exposes the default ports.

Reply:
Another example is Prometheus' or Grafana Agent's Docker Service Discovery. The exposed Port will be used to scrape metrics. If you don't expose (or map) your port, the service discovery doesn't scrape the container.

Reply:
Thanks @durcon, I added that to the motivations list.